test/util/framework/per_test_framework.go

```diff
@@ -236,16 +263,67 @@ func (tc *perItOrDescribeTestContext) NewResourceGroup(ctx context.Context, reso
 // cleanupResourceGroup is the standard resourcegroup cleanup.  It attempts to
 // 1. delete all HCP clusters and wait for success
 // 2. delete the resource group and wait for success
-func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string) error {
+func cleanupResourceGroup(ctx context.Context, hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string, timeout time.Duration) error {
 	errs := []error{}
 
 	ginkgo.GinkgoLogr.Info("deleting all hcp clusters in resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteAllHCPClusters(ctx, hcpClient, resourceGroupName, timeout); err != nil {
 		return fmt.Errorf("failed to cleanup resource group: %w", err)
 	}
 
 	ginkgo.GinkgoLogr.Info("deleting resource group", "resourceGroup", resourceGroupName)
-	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, 60*time.Minute); err != nil {
+	if err := DeleteResourceGroup(ctx, resourceGroupsClient, resourceGroupName, false, timeout); err != nil {
+		return fmt.Errorf("failed to cleanup resource group: %w", err)
+	}
+
+	return errors.Join(errs...)
+}
+
+// cleanupResourceGroupNoRP performs cleanup when the resource provider is not available.
+// This is used to cleanup personal dev e2e test runs, where the infra is already gone so there's no
+// RP to call for HCP deletion.
+//  1. discovers any "managed" resource groups whose ManagedBy references a resource in the parent
+//     resource group and deletes them (using 'force' to speed up VM/VMSS deletion).
+//  2. deletes the parent resource group itself.
+func cleanupResourceGroupNoRP(ctx context.Context, resourceGroupsClient *armresources.ResourceGroupsClient, resourceGroupName string, timeout time.Duration) error {
```

Comment:
Ideally, we still try to delete the HCP first - the RP may well be running & healthy, and we want to test that cleanup works as expected. The backstop of removing managed groups should trigger some kind of trackable error we can see so we debug why normal cleanup didn't work

Reply:
By the way, I've changed the semantics of the switch to choose one deletion flow or the other. I realized it was misleading. Hopefully now the mode names convey better their purpose.